### ISSUE TITLE
Incorrect returned value type in several functions

### ISSUE THREAD

**Issue by fdur** [eclipse/wakaama#314]
In `core/block1.c`:
```
coap_status_t coap_block1_handler(lwm2m_block1_data_t ** pBlock1Data,
                                  uint16_t mid,
                                  uint8_t * buffer,
                                  size_t length,
                                  uint16_t blockSize,
                                  uint32_t blockNum,
                                  bool blockMore,
                                  uint8_t ** outputBuffer,
                                  size_t * outputLength)
```

in `core/er-coap-13/er-coap-13.h`:
```
/* CoAP response codes */
typedef enum {
  NO_ERROR = 0,
  ...
  INTERNAL_SERVER_ERROR_5_00 = 160,     /* INTERNAL_SERVER_ERROR */
  ...
} coap_status_t;
```
in `core/liblwm2m.h`:
`#define COAP_500_INTERNAL_SERVER_ERROR  (uint8_t)0xA0`

`coap_status_t` is a enum but values like `COAP_500_INTERNAL_SERVER_ERROR` are returned (defined value) (even if the value is correct)

Impacted functions:
`core/block1.c`: `coap_block1_handler`
`core/bootstrap.c`: `bootstrap_handleFinish`, `bootstrap_handleCommand`, `prv_checkServerStatus`, `bootstrap_handleDeleteAll`
`core/management.c`: `dm_handleRequest`
`core/objects.c`: `object_readData`, `object_write`, `object_execute`, `object_create`, `object_delete`, `object_delete`, `object_writeInstance`
`core/observe.c`: `observe_handleRequest`, `observe_setParameters`
`core/packet.c`: `handle_request`, `lwm2m_handle_packet`

**Comment by davidgraeff** [comment]
That's one reason why I thing there should be the -pedantic -Wall, maybe even -Werror, switches enabled for compiling and a continuous integration service like travis be used.

**Comment by dnav** [comment]
Proposed fix in #323

**Comment by dnav** [comment]
Fixed by #323
